nextcord/polls.py

```diff
@@ -0,0 +1,437 @@
+
+    def __init__(self, data: PollResultsPayload, poll: Poll) -> None:
+        self.is_finalized: bool = data["is_finalized"]
+        self.answer_counts: List[PollAnswerCount] = [
+            PollAnswerCount(answer_count, poll) for answer_count in data["answer_counts"]
+        ]
+        self.poll: Poll = poll
+
+    def to_dict(self) -> PollResultsPayload:
+        payload: PollResultsPayload = {
+            "is_finalized": self.is_finalized,
+            "answer_counts": [answer_count.to_dict() for answer_count in self.answer_counts],
+        }
+        return payload
+
+
+class PollCreateRequest:
+    """
+    A poll create request.
+    You must use this to create a poll.
+
+    .. versionadded:: 3.0
+
+    Attributes
+    ----------
+    question: :class:`PollMedia`
+        The question of the poll. Current only `text` is supported.
+    duration: :class:`int`
+        The number of hours this poll should be open for.
+    allow_multiselect: :class:`bool`
+        Whether voters can select more than one choice.
+    answers: Optional[List[:class:`PollAnswer`]]
+        The answers to the poll.
+        If omitted then it must be added later through :func:`PollCreateRequest.add_answer`.
+    layout_type: Optional[:class:`PollLayoutType`]
+        The layout type of the poll. Defaults to :attr:`PollLayoutType.default`
+    """
+
+    __slots__ = (
+        "question",
+        "answers",
+        "duration",
+        "allow_multiselect",
+        "layout_type",
+    )
+
+    def __repr__(self) -> str:
+        return (
+            "<PollCreateRequest "
+            f"question={self.question!r} "
+            f"answers={self.answers!r} "
+            f"duration={self.duration!r} "
+            f"allow_multiselect={self.allow_multiselect!r} "
+            f"layout_type={self.layout_type!r} "
+            ">"
+        )
+
+    def __init__(
+        self,
+        question: PollMedia,
+        duration: int,
+        allow_multiselect: bool,
+        layout_type: Optional[PollLayoutType] = MISSING,
+        answers: Optional[List[PollAnswer]] = None,
+    ) -> None:
+        self.question: PollMedia = question
+        if answers:
+            self.answers: List[PollAnswer] = answers
+        else:
+            self.answers: List[PollAnswer] = []
+        self.duration: int = duration
+        self.allow_multiselect: bool = allow_multiselect
+        self.layout_type: Optional[PollLayoutType] = layout_type
+
+    def to_dict(self) -> PollCreateRequestPayload:
+        payload: PollCreateRequestPayload = {
+            "question": self.question.to_dict(),
+            "answers": [answer.to_dict() for answer in self.answers],
+            "duration": self.duration,
+            "allow_multiselect": self.allow_multiselect,
+        }
+        if self.layout_type:
+            payload["layout_type"] = self.layout_type.value
+        return payload
+
+    def add_answer(self, text: str, emoji: Optional[EmojiInputType] = None) -> Self:
```

Comment:
Im not sure, why this returns `Self`? It may be `None` here.

Reply:
Yes, I am confused about why you are doing this. Why does the method that adds `Answer` return itself? You may return None.

Reply:
Aaah, I didn't notice, although I don't like it anyway xD